lib/elixir/lib/uri.ex

```diff
@@ -279,10 +280,10 @@ defmodule URI do
       "a str%69ng"
 
   """
-  @spec encode(binary, (byte -> boolean)) :: binary
-  def encode(string, predicate \\ &char_unescaped?/1)
-      when is_binary(string) and is_function(predicate, 1) do
-    for <<char <- string>>, into: "", do: percent(char, predicate)
+  @spec encode(binary, as_boolean(char)) :: binary
```

Comment:
```suggestion
  @spec encode(String.t(), (byte -> as_boolean(term)) :: String.t()
```

Reply:
It should be actually `(char -> as_boolean(term))`

Reply:
I think it's `byte` 🙂 

```elixir
iex(1)> URI.encode("🙂", &IO.inspect/1)
240
159
153
130
"🙂"
```